addOns/spider/src/main/java/org/zaproxy/addon/spider/parser/SpiderSitemapXmlParser.java

```diff
@@ -48,23 +50,25 @@ public class SpiderSitemapXmlParser extends SpiderParser {
     /** a pattern to match the sitemap.xml file name */
     private static final Pattern SITEMAP_XML_FILENAME_PATTERN = Pattern.compile("/sitemap\\.xml$");
 
-    /** a pattern to match the sitemap.xml file.. hint: It's XML */
-    private static final Pattern XML_PATTERN =
-            Pattern.compile(
-                    "^<\\?xml\\s+version\\s*=\\s*\"[0-9.]+\"\\s+encoding\\s*=\\s*\"[^\"]+\"\\s*\\?>");
+    private static final String PARSE_FAIL_WARNING =
+            "An error occurred trying to parse sitemap.xml";
 
     /** used to parse the XML based file format */
     private static DocumentBuilder dBuilder;
 
     /** an x path expression to match the "loc" tag in sitemap.xml */
     private static XPathExpression xpathLocationExpression;
 
+    /** an x path expression to match xhtml:link hrefs in sitemap.xml */
+    private static XPathExpression xpathXhtmlLinkHrefExpression;
+
     /** statically initialise the XML DocumentBuilderFactory and DocumentBuilder */
     static {
         try {
             dBuilder = XmlUtils.newXxeDisabledDocumentBuilderFactory().newDocumentBuilder();
             XPath xpath = XPathFactory.newInstance().newXPath();
             xpathLocationExpression = xpath.compile("/urlset/url/loc/text()");
+            xpathXhtmlLinkHrefExpression = xpath.compile("//*[@href or @HREF]");
```

Comment:
I tried a bunch of things for more specific expressions and had zero luck, if anyone has ideas on this I'd be happy to adjust.